### ISSUE TITLE
How to allow inheritance of swapped model?

### ISSUE THREAD

**Issue by angelkenneth** [openwisp/django-swappable-models#16]
Hi, this is from a question of mine that i posted at StackOverflow.  Since nobody has answered in a while, i thought that i should asked it here also.

Excerpt from [StackOverflow](http://stackoverflow.com/questions/43971324/django-how-to-allow-inheritance-of-swappable-model)

**Django - How to allow inheritance of swappable model?**

I am using [swappable](https://github.com/wq/django-swappable-models) to make a reusable app (named `Meat`) that provides models that Developers can swap for their own.  That model is a super class of other models.

    from django.db.models import Model, CharField
    from swapper import swappable_setting
    
    class AbstractMeat(Model):
        class Meta:
            abstract = True
        name = CharField(max_length=16)
    
    class Meat(AbstractMeat):
        class Meta:
            swappable = swappable_setting("cyber", "Meat")
    
    class Pork(Meat):
        pass
    
    class Fish(Meat):
        pass

To test this, i created the `real` app and set `MEAT_MEAT_MODEL`.

    # settings.py
    MEAT_MEAT_MODEL = "real.RealMeat"
    
    # real/models.py
    from django.forms import IntegerField
    from cyber.models import AbstractMeat
    
    class RealMeat(AbstractMeat):
        price = IntegerField()

Running `runserver` i get this error:

    meat.Fish.meat_ptr: (fields.E301) Field defines a relation with the model 'meat.Meat', which has been swapped out.
    	HINT: Update the relation to point at 'settings.MEAT_MEAT_MODEL'.
    meat.Pork.meat_ptr: (fields.E301) Field defines a relation with the model 'meat.Meat', which has been swapped out.
    	HINT: Update the relation to point at 'settings.MEAT_MEAT_MODEL'.

This error arises on Django 1.9 to 1.11, but for my purpose only 1.11 is critical.

I tried overriding the `meat_ptr` as instructed in [Multi-table inheritance](https://docs.djangoproject.com/en/1.11/topics/db/models/#multi-table-inheritance) like so:

    from swapper import get_model_name
    from django.db.models import OneToOneField, CASCADE
    
    class Pork(Meat):
        meat_ptr = OneToOneField(
            get_model_name("meat", "Meat"), CASCADE,
            parent_link=True)

But it gives me this error on 1.11 and 1.10 (but not 1.9):

    django.core.exceptions.FieldError: Auto-generated field 'meat_ptr' in class 'Pork' for parent_link to base class 'Meat' clashes with declared field of the same name.

In conclusion, how do i make this happen?

**Comment by nemesisdesign** [comment]
Closing because this is not a bug.
